lib/dynflow/watchers/memory_consumption_watcher.rb

```diff
@@ -0,0 +1,32 @@
+require 'get_process_mem'
+
+module Dynflow
+  module Watchers
+    class MemoryConsumptionWatcher
+
+      attr_reader :memory_limit, :world
+
+      def initialize(world, memory_limit, options)
```

Comment:
This is why I have added `options[:initial_wait]` you can pass a longer value in order to give the executor the time it needs to start.